src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -720,25 +720,10 @@ public void InsertDefault()
         }
 
         [Fact]
-        public void InsertRangeNoOpIdentity()
-        {
-            Assert.Equal(s_empty, s_empty.InsertRange(0, s_empty));
-            Assert.Equal(s_oneElement, s_empty.InsertRange(0, s_oneElement)); // struct overload
-            Assert.Equal(s_oneElement, s_empty.InsertRange(0, (IEnumerable<int>)s_oneElement)); // enumerable overload
-            Assert.Equal(s_oneElement, s_oneElement.InsertRange(0, s_empty));
-        }
-
-        [Fact]
-        public void InsertRangeEmpty()
+        public void InsertRangeEmptyInvalid()
         {
             Assert.Throws<NullReferenceException>(() => s_emptyDefault.Insert(-1, 10));
             Assert.Throws<NullReferenceException>(() => s_emptyDefault.Insert(1, 10));
-            Assert.Equal(new int[0], s_empty.InsertRange(0, Enumerable.Empty<int>()));
-            Assert.Equal(s_empty, s_empty.InsertRange(0, Enumerable.Empty<int>()));
-            Assert.Equal(new[] { 1 }, s_empty.InsertRange(0, new[] { 1 }));
-            Assert.Equal(new[] { 2, 3, 4 }, s_empty.InsertRange(0, new[] { 2, 3, 4 }));
-            Assert.Equal(new[] { 2, 3, 4 }, s_empty.InsertRange(0, Enumerable.Range(2, 3)));
-            Assert.Equal(s_manyElements, s_manyElements.InsertRange(0, Enumerable.Empty<int>()));
```

Comment:
Similar here. Either you're removing interesting tests, or you're refactoring them. Refactoring tests at the same time as introducing a product change makes it much more difficult to ascertain that you haven't changed anything that would have caused a pre-existing test to fail. I wonder if we should revert your test removes and changes, and just add your new tests, then in a subsequent PR do any refactoring you'd like to do separately?